src/Routed/SingleReviewPage.jsx

```diff
@@ -0,0 +1,78 @@
+import { Link } from "react-router-dom";
+import { useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { getSingleReviewById, getCommentsBySingleReviewId } from "../Apis";
+import SingleReviewCommentList from "../Components/SingleReviewCommentList";
+
+export default function SingleReviewPage() {
+  const { id } = useParams();
+
+  const [reviewObj, setReviewObj] = useState({
+    review_id: 0,
```

Comment:
Unless there's a reason that you've identified for doing it this way, I think useState({}) would be sufficient here!

Reply:
Thanks for the feedback!

About the useState object:

My initial idea was to use useState() for each value, but then I thought it would be too lengthy to do so many [useState, setState] pairs when I could do only one for the whole object and then access the properties as I needed them. 

But then when I was refactoring the Categories Page it started throwing an error, saying something like 'cannot read properties of undefined, reading 0', it took me ages to figure this one out because once I started console logging and doing changes to the function it would actually work, but then it would crash once I reloaded the page.

And then I finally realised that was where the issue laid, the 'first' rendering was trying to read a property / method of a value that was part of a property of categoryObj that didn't exist yet (neither the key or the value), because at that stage it was an empty object. So I pre-added those properties as placeholders to avoid that error, in both Categories and SingleReview Pages.